adblock-lean

```diff
@@ -915,10 +923,122 @@ update()
 	else
 		log_msg -err "Unable to download latest version of adblock-lean."
 	fi
-        rm -f /var/run/adblock-lean/adblock-lean.latest /var/run/adblock-lean/uclient-fetch_err
+	rm -f /var/run/adblock-lean/adblock-lean.latest /var/run/adblock-lean/uclient-fetch_err
+}
+
+mk_lock() {
+	check_lock
+	case ${?} in
+		1) exit 1 ;;
+		2)
+			report_pid_action
+			log_msg "Refusing to open another instance."
+			exit 0
+	esac
+
+	if [ -n "${pid_file}" ]
+	then
+		mkdir -p "${pid_file%/*}" || { log_msg -err "Error: Failed to create directory '${pid_file%/*}'."; exit 1; }
+		printf '%s\n' "${$} ${action}" > "${pid_file}" || { log_msg -err "Error: Failed to write to pid file '${pid_file}'."; exit 1; }
+	else
+		log_msg -err "Internal error: \${pid_file} variable is unset."
+		exit 1
+	fi
+	:
 }
 
-if [ "${action}" != "help" ] && [ "${action}" != "gen_config" ]
+rm_lock() {
```

Comment:
Would you mind amending the functions to: 

x()
{
…
}

Reply:
> Would you mind amending the functions to

Done